cmd/tfchaind/main.go

```diff
@@ -26,6 +27,12 @@ func main() {
 
 // SetupNetworks injects the correct chain constants and genesis nodes based on the chosen network
 func SetupNetworks(name string) (daemon.NetworkConfig, error) {
+	// configure any tfchain-specific logic related to how
+	// unlock conditions/fulfillments are to be interpreted/used.
+	ConfigureUnlockConditions(name)
```

Comment:
I would drop  this top level method that includes a new switch on network name  and shift it in to the switch below.

Reply:
Sure, sounds reasonable. Please review this PR in Rivine than ASAP: https://github.com/rivine/rivine/pull/375, it is required to make your new top-level function make reality. Once that is accepted I'll merge those rivine changes over to this branch as well.